theano/gpuarray/dnn.py

```diff
@@ -2833,6 +2833,289 @@ def local_abstractconv3d_cudnn_graph(op, context_name, inputs, outputs):
     return [rval]
 
 
+class GpuDnnTransformerDesc(COp):
+    """
+    This Op builds a spatial transformer descriptor for use in spatial transformer network
+    operations.
+    """
+    __props__ = ('precision',)
+    params_type = ParamsType(precision=cudnn.cudnnDataType_t)
+
+    def c_headers(self):
+        return ['cudnn.h', 'cudnn_helper.h']
+
+    def c_header_dirs(self):
+        header_dirs = [os.path.dirname(__file__)]
+        if config.dnn.include_path:
+            header_dirs += [config.dnn.include_path]
+        return header_dirs
+
+    def c_libraries(self):
+        return ['cudnn']
+
+    def c_lib_dirs(self):
+        lib_dirs = []
+        if config.dnn.library_path:
+            lib_dirs += [config.dnn.library_path]
+        return lib_dirs
+
+    def do_constant_folding(self, node):
+        return False
+
+    def __init__(self, precision=theano.config.floatX):
+        COp.__init__(self, ["c_code/dnn_sptf_desc.c"], "APPLY_SPECIFIC(dnn_sptf_desc)")
+        assert cudnn.cudnnDataType_t.has_alias(precision)
+        self.precision = precision
+
+    def make_node(self, dimensions):
+        dimensions = as_tensor_variable(dimensions)
+        assert dimensions.dtype in theano.tensor.basic.int_dtypes
+        assert dimensions.ndim == 1
+        dimensions = theano.tensor.basic.cast(dimensions, 'int64')
+
+        node = Apply(self, [dimensions],
+                     [CDataType("cudnnSpatialTransformerDescriptor_t",
+                      freefunc="cudnnDestroySpatialTransformerDescriptor")()])
+        # DebugMode cannot compare the values of CDataType variables, so by
+        # default it returns False all the time. To prevent DebugMode from
+        # complaining because of the MergeOptimizer, we make this variable
+        # always compare to True.
+        out = node.outputs[0]
+        out.tag.values_eq_approx = tensor.type.values_eq_approx_always_true
+        return node
+
+    def c_code_cache_version(self):
+        return (super(GpuDnnTransformerDesc, self).c_code_cache_version(), version())
+
+
+class GpuDnnTransformerGrid(DnnBase):
+    __props__ = ()
+    _cop_num_inputs = 3
+    _cop_num_outputs = 1
+    _f16_ok = True
+
```

Comment:
Also add `check_input = False` here. It will help Theano to not generate two many variants of C code, as we either force types (e.g. with int64) or don't use any macro related to inputs or outputs dtypes into our C codes.